dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -391,8 +454,18 @@ public Service initializeServiceFromGitHub(String repositoryId, String subclass)
      * @param workflowName Name of the workflow
      * @return Workflow
      */
-    public BioWorkflow initializeWorkflowFromGitHub(String repositoryId, String subclass, String workflowName) {
+    public Workflow initializeWorkflowFromGitHub(String repositoryId, String subclass, String workflowName) {
         BioWorkflow workflow = new BioWorkflow();
+        return setWorkflowInfo(repositoryId, subclass, workflowName, workflow);
+    }
+
+    public Workflow initializeOneStepWorkflowFromGitHub(String repositoryId, String subclass, String workflowName) {
+        OneStepWorkflow workflow = new OneStepWorkflow();
+        return setWorkflowInfo(repositoryId, subclass, workflowName, workflow);
```

Comment:
You don't validate that it's a one step workflow, e.g., I think I could register a multi-step workflow and it would work.

Reply:
This feature is ultimately for CWL and eventually Galaxy tools since Nextflow and WDL both don't have the concept of a tool (although we made one up for WDL). Eventually the version gets validated and will say whether it's a tool or not.https://github.com/dockstore/dockstore/pull/4321/files#diff-fc522f43938a138dbd009c313899114789e249866f7b03c0919b1a8839b1c24aR698 so it's as good as whatever our current validation is.